Compiler warnings Mac Ventura

[ 12%] Building CXX object CMakeFiles/WorldBuilder.dir/Unity/unity_0_cxx.cxx.o
[ 12%] Building CXX object CMakeFiles/WorldBuilder.dir/source/world_builder/parameters.cc.o
In file included from /Applications/GWB/WorldBuilder/build/CMakeFiles/WorldBuilder.dir/Unity/unity_0_cxx.cxx:231:
/Applications/GWB/WorldBuilder/source/world_builder/utilities.cc:1198:46: warning: argument type 'int *' doesn't match specified 'MPI' type tag that requires 'unsigned int *' [-Wtype-safety]
                  const int ierr = MPI_Bcast(&invalid_file_size, 1, MPI_UNSIGNED, 0, comm);
                                             ^~~~~~~~~~~~~~~~~~     ~~~~~~~~~~~~
/Applications/GWB/WorldBuilder/source/world_builder/utilities.cc:1213:46: warning: argument type 'int *' doesn't match specified 'MPI' type tag that requires 'unsigned int *' [-Wtype-safety]
                  const int ierr = MPI_Bcast(&invalid_file_size, 1, MPI_UNSIGNED, 0, comm);
                                             ^~~~~~~~~~~~~~~~~~     ~~~~~~~~~~~~
/Applications/GWB/WorldBuilder/source/world_builder/utilities.cc:1225:36: warning: argument type 'int *' doesn't match specified 'MPI' type tag that requires 'unsigned int *' [-Wtype-safety]
              int ierr = MPI_Bcast(&filesize, 1, MPI_UNSIGNED, 0, comm);
                                   ^~~~~~~~~     ~~~~~~~~~~~~
/Applications/GWB/WorldBuilder/source/world_builder/utilities.cc:1240:36: warning: argument type 'int *' doesn't match specified 'MPI' type tag that requires 'unsigned int *' [-Wtype-safety]
              int ierr = MPI_Bcast(&filesize, 1, MPI_UNSIGNED, 0, comm);
                                   ^~~~~~~~~     ~~~~~~~~~~~~

--- a/source/world_builder/utilities.cc
+++ b/source/world_builder/utilities.cc
@@ -1195,7 +1195,7 @@ namespace WorldBuilder
               if (!filestream)
                 {
                   // broadcast failure state, then throw
-                  const int ierr = MPI_Bcast(&invalid_file_size, 1, MPI_UNSIGNED, 0, comm);
+                  const int ierr = MPI_Bcast(&invalid_file_size, 1, MPI_INT, 0, comm);
                   WBAssertThrow (ierr,
                                  std::string("Could not open file <") + filename + ">.");
                 }
@@ -1210,7 +1210,7 @@ namespace WorldBuilder
               catch (const std::ios::failure &)
                 {
                   // broadcast failure state, then throw
-                  const int ierr = MPI_Bcast(&invalid_file_size, 1, MPI_UNSIGNED, 0, comm);
+                  const int ierr = MPI_Bcast(&invalid_file_size, 1, MPI_INT, 0, comm);
                   WBAssertThrow(ierr == 0, "MPI_Bcast failed.");
                   WBAssertThrow (false,
                                  std::string("Could not read file content from <") + filename + ">.");
@@ -1222,7 +1222,7 @@ namespace WorldBuilder
               filesize = static_cast<int>(data_string.size());
 
               // Distribute data_size and data across processes
-              int ierr = MPI_Bcast(&filesize, 1, MPI_UNSIGNED, 0, comm);
+              int ierr = MPI_Bcast(&filesize, 1, MPI_INT, 0, comm);
               WBAssertThrow(ierr == 0, "MPI_Bcast failed.");
 
               // Receive and store data
@@ -1237,7 +1237,7 @@ namespace WorldBuilder
             {
               // Prepare for receiving data
               int filesize = 0;
-              int ierr = MPI_Bcast(&filesize, 1, MPI_UNSIGNED, 0, comm);
+              int ierr = MPI_Bcast(&filesize, 1, MPI_INT, 0, comm);
               WBAssertThrow(ierr == 0, "MPI_Bcast failed.");
               WBAssertThrow(filesize != -1,
                             std::string("Could not open file <") + filename + ">.");

